services/QuillLMS/app/workers/snapshots/cache_snapshot_top_x_worker.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Snapshots
+  class CacheSnapshotTopXWorker
```

Comment:
A ton of this code is just a copy of the `CacheSnapshotCountWorker`, and the duplication makes me a little uneasy.  But I figured doing it twice might be fine?  We also don't have a real history of inherited class hierarchies in worker code, so I was hesitant to refactor into a parent with two children.  Again, super open to suggestions.

Reply:
The `perform` method is nearly identical, so we could abstract that into a parent Worker class

Reply:
I usually wait to abstract classes until I have three examples.  Two can be premature.